components/rmrk/Gallery/Flipper.vue

```diff
@@ -0,0 +1,152 @@
+<template>
+  <Holder
+    :table-rows-option="ownerEventsOfNft"
+    group-key-option="Flipper"
+    openOnDefault
+    date-header-label="Last Activity"
+    name-header-label="User"
+    sale-header-label="Sold"
+    :collapseTitleOption="$t('Flipper')"
+    hideCollapse />
+</template>
+
+<script lang="ts">
+import PrefixMixin from '@/utils/mixins/prefixMixin'
+import { Component, Prop, Watch, mixins } from 'nuxt-property-decorator'
+import { Interaction } from '../service/scheme'
+import { TableRow } from '@/components/rmrk/Gallery/Holder/Holder.vue'
+import { parseDate } from '@/components/rmrk/Gallery/Holder/helper'
+import { formatDistanceToNow } from 'date-fns'
+
+const components = {
+  Holder: () => import('@/components/rmrk/Gallery/Holder/Holder.vue'),
+}
+
+type FlipperTableRowMap = Record<string, TableRow[]>
+
+@Component({ components })
+export default class Flipper extends mixins(PrefixMixin) {
+  @Prop({ type: Array }) events!: Interaction[]
+  private ownerEventsOfNft: TableRow[] = []
+
+  createNFTList() {
+    if (this.events.length) {
+      this.ownerEventsOfNft = this.generateNFTList()
+    }
+  }
+
+  private generateNFTList(): TableRow[] {
+    const rowListMap: FlipperTableRowMap = {}
+
+    for (const newEvent of this.events) {
+      const date = new Date(newEvent['timestamp'])
+      const timestamp = date.getTime()
+      const dateStr = parseDate(date)
+      const formatTime = formatDistanceToNow(date, { addSuffix: true })
+      const block = String(newEvent['blockNumber'])
+      const collectionId = newEvent['nft']['collection']['id']
+      const commonInfo = {
+        Date: dateStr,
+        Time: formatTime,
+        SortKey: timestamp,
+        Timestamp: timestamp,
+        Block: block,
+        CollectionId: collectionId,
+        Amount: 1,
+      }
+      const nftId = newEvent['nft'].id
+      rowListMap[nftId] = rowListMap[nftId] ?? []
```

Comment:
What could we do on rubick so we do not need to put spaghetti of code for each table?

Reply:
Maybe there is a better way to do it.

I can think of for the time being is to do the same process on the server-side and cache the result, so that avoid the heavy process in the client : )

Reply:
Please [open a new issue in rubick](https://github.com/kodadot/rubick/issues/new)

Reply:
Done